### PR TITLE
Update arcade version to unblock signed build

### PR DESCRIPTION
**Pull request by Cosifne**
Update arcade version to unblock us from the error in SDL stage.

I have a test run here https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4412680&view=results

### CODE REVIEW

**Review by JoeRobich — APPROVED** [review]

File: eng/Version.Details.xml [hunk 1]
```diff
@@ -7,9 +7,9 @@
       <Uri>https://github.com/dotnet/arcade</Uri>
       <Sha>620462dd065490620c03a80cea8e251004f6bf21</Sha>
     </Dependency>
-    <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="3.9.0-4.20631.6">
+    <Dependency Name="Microsoft.Net.Compilers.Toolset" Version="3.9.0-4.21076.10">
       <Uri>https://github.com/dotnet/roslyn</Uri>
-      <Sha>c34bb049f1f74256f109794509cd9f0f1c23faea</Sha>
```
Comment (RikkiGibson) [inline eng/Version.Details.xml:12]: It looks like only the compilers toolset is updated here, not the arcade version.

Comment (Cosifne) [reply eng/Version.Details.xml:12]: I just cherry-pick the change from here https://github.com/dotnet/roslyn/pull/50828
Do you feel the update of compiler toolset is not needed?

Comment (RikkiGibson) [reply eng/Version.Details.xml:12]: I think it's OK to update the toolset. I mentioned this to make sure of whether it's expected to have an arcade update in this PR.

**Comment by JoeRobich** [comment]
Merging to unblock signed builds.

**Pull request merged**